crates/sui-swarm-config/src/network_config_builder.rs

```diff
@@ -458,3 +459,354 @@ impl<R: rand::RngCore + rand::CryptoRng> ConfigBuilder<R> {
+                external_address: Some(utils::socket_address_to_udp_multiaddr(address)),
+                seed_peers,
+                ..Default::default()
+            }
+        };
+
+        let rpc_port = self.rpc_port.unwrap_or_else(|| get_available_port(9000));
+        let jsonrpc_server_url = format!("{}:{}", listen_ip, rpc_port);
+        let json_rpc_address: SocketAddr = jsonrpc_server_url.parse().unwrap();
+
+        let supported_protocol_versions = match &self.supported_protocol_versions_config {
+            ProtocolVersionsConfig::Default => SupportedProtocolVersions::SYSTEM_DEFAULT,
+            ProtocolVersionsConfig::Global(v) => *v,
+            ProtocolVersionsConfig::PerValidator(func) => func(0, None),
+        };
+
+        Ok(NodeConfig {
+            protocol_key_pair: AuthorityKeyPairWithPath::new(protocol_key_pair),
+            account_key_pair: KeyPairWithPath::new(SuiKeyPair::Ed25519(account_key_pair)),
+            worker_key_pair: KeyPairWithPath::new(SuiKeyPair::Ed25519(worker_key_pair)),
+            network_key_pair: KeyPairWithPath::new(SuiKeyPair::Ed25519(network_key_pair)),
+
+            db_path: db_path.join(dir_name),
+            network_address,
+            metrics_address: utils::available_local_socket_address(),
+            // TODO: admin server is hard coded to start on 127.0.0.1 - we should probably
+            // provide the entire socket address here to avoid confusion.
+            admin_interface_port: self.admin_port.unwrap_or_else(|| get_available_port(8888)),
+            json_rpc_address,
+            consensus_config: None,
+            enable_event_processing: self.enable_event_store,
+            enable_index_processing: default_enable_index_processing(),
+            genesis: validator_config.genesis.clone(),
+            grpc_load_shed: None,
+            grpc_concurrency_limit: None,
+            p2p_config,
+            authority_store_pruning_config: AuthorityStorePruningConfig::fullnode_config(),
+            end_of_epoch_broadcast_channel_capacity:
+                default_end_of_epoch_broadcast_channel_capacity(),
+            checkpoint_executor_config: Default::default(),
+            metrics: None,
+            supported_protocol_versions: Some(supported_protocol_versions),
+            db_checkpoint_config: self.db_checkpoint_config,
+            indirect_objects_threshold: usize::MAX,
+            // Copy the expensive safety check config from the first validator config.
+            expensive_safety_check_config: validator_config.expensive_safety_check_config.clone(),
+            name_service_resolver_object_id: None,
+            transaction_deny_config: Default::default(),
+            certificate_deny_config: Default::default(),
+            state_debug_dump_config: Default::default(),
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::collections::HashSet;
+    use std::sync::Arc;
+    use sui_adapter::execution_mode;
```

Comment:
AFAICT, these tests are the only use of the `sui-adapter` and `sui-move-natives` crates from this crate.  Can we try making those `dev-dependencies`?